Data/models/DataController.swift

```diff
@@ -44,6 +40,7 @@ public class DataController: NSObject {
                 }
             } catch {
                 log.error(error)
+                throw error
```

Comment:
Let's just let this auto-throw and move error logging up a level, no real point here of catching and throwing the same error.